src/bms/model/control_flow.rs

```diff
@@ -0,0 +1,631 @@
+        self.branches.push(branch);
+        self
+    }
+
+    /// Number of branches.
+    #[must_use]
+    pub const fn len(&self) -> usize {
+        self.branches.len()
+    }
+
+    /// Returns true if there are no branches in this random block.
+    #[must_use]
+    pub const fn is_empty(&self) -> bool {
+        self.branches.is_empty()
+    }
+
+    /// Get branch by index.
+    #[must_use]
+    pub fn at(&self, index: usize) -> Option<&IfBlock<'a>> {
+        self.branches.get(index)
+    }
+
+    /// Get mutable branch by index.
+    pub fn at_mut(&mut self, index: usize) -> Option<&mut IfBlock<'a>> {
+        self.branches.get_mut(index)
+    }
+
+    /// Convert the model into lex tokens representing the random block.
+    #[must_use]
+    pub fn into_tokens(self) -> Vec<Token<'a>> {
+        let mut out = Vec::new();
+        match &self.value {
+            ControlFlowValue::GenMax(max) => out.push(Token::Header {
+                name: "RANDOM".into(),
+                args: max.to_string().into(),
+            }),
+            ControlFlowValue::Set(val) => out.push(Token::Header {
+                name: "SETRANDOM".into(),
+                args: val.to_string().into(),
+            }),
+        }
+
+        self.branches.into_iter().for_each(|branch| {
+            // Emit head IF
+            out.push(Token::Header {
+                name: "IF".into(),
+                args: branch.condition.to_string().into(),
+            });
+            out.extend(
+                branch
+                    .head_units
+                    .into_iter()
+                    .flat_map(TokenUnit::into_tokens),
+            );
+
+            // Emit chained ELSEIF/ELSE entries
+            let mut node = branch.chain;
+            loop {
+                match node {
+                    IfChainEntry::ElseIf { cond, units, next } => {
+                        out.push(Token::Header {
+                            name: "ELSEIF".into(),
+                            args: cond.to_string().into(),
+                        });
+                        out.extend(units.into_iter().flat_map(TokenUnit::into_tokens));
+                        node = *next;
+                    }
+                    IfChainEntry::Else { units } => {
+                        out.push(Token::Header {
+                            name: "ELSE".into(),
+                            args: "".into(),
+                        });
+                        out.extend(units.into_iter().flat_map(TokenUnit::into_tokens));
+                        break;
+                    }
+                    IfChainEntry::EndIf => break,
+                }
+            }
+
+            // Close the IF-chain
+            out.push(Token::Header {
+                name: "ENDIF".into(),
+                args: "".into(),
+            });
+        });
+
+        out.push(Token::Header {
+            name: "ENDRANDOM".into(),
+            args: "".into(),
+        });
```

Comment:
`Token::header` is useful to get a literal token. Also same as it at `Switch::into_tokens`.
```suggestion
                        out.push(Token::header("ELSE", ""));
                        out.extend(units.into_iter().flat_map(TokenUnit::into_tokens));
                        break;
                    }
                    IfChainEntry::EndIf => break,
                }
            }

            // Close the IF-chain
            out.push(Token::header("ENDIF", ""));
        });

        out.push(Token::header("ENDRANDOM", ""));
```